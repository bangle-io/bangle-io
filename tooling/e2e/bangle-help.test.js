const { sleep, url } = require('./helpers');

beforeEach(async () => {
  await jestPuppeteer.resetPage();
  await page.goto(url);
  page.on('error', (err) => {
    console.log('error happen at the page');
    throw err;
  });
  page.on('pageerror', (pageerr) => {
    console.log('pageerror occurred');
    throw pageerr;
  });
  await page.evaluate(() => localStorage.clear());
});

test('Landing page is correct', async () => {
  await sleep(500);
  const handle = await page.$('.main-content');
  const result = await handle.evaluate((node) => node.innerText);
  expect(result.includes('bangle.io')).toBe(true);
  expect(result).toMatchSnapshot();
});
