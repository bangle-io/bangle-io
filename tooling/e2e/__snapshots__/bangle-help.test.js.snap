// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing page is correct 1`] = `
"bangle.io

(beta)

A local note taking app for the rest of us 🐥

Hello,

Thanks for taking out the time to read about bangle.io. I know it is an insanely crowded market and every day a new note taking app pops up. Yet here I am, passionately working on this project for the past 3 years hoping people will like it.

(💡 yes you can edit this page!)

1. You own your notes

Local: Modify notes right from your computer. Think Notion, but local only.

No data lock-in: bangle stores your notes in a human readable Markdown format.

2. Not your usual note taking app

WYSIWYG: Even though we save things in Markdown format, Bangle will enrich your eyes with rich text formatting.

Modern rich text editor: Bangle uses its sister project bangle.dev which unlocks us to build a powerful editor like Notion or Dropbox paper and support real time collaboration.

Extensibility: Even though it's a web app it was built ground up to be extended by awesome extensions. (We have bunch of community extensions coming out soon.)

3. Portability at its core

No Electron:  ❤️ for web and speed, thats it.

Performant: Throw thousands of notes at it and Bangle will chug along fine. (Try opening a heavy project like 10,000 Markdown Files.)

Why should you try it out?

The cost of trying it out is low as this is a web app, not installation and as I mentioned above there is no data lock-in. You can open your existing Markdown notes try it out and still continue to use your favorite note taking app.

Bangle.io is in its early stages, so expect some important features missing and some stability issues. If you are already happy with your current setup, thats great, but if you think there is a potential here please help us spread the word, every bit counts 🙋‍♀️ !

Head to 📗 [[Getting Started]] to create your workspace

Follow the project on 🐦 Twitter.

For thoughts or ❤️ please join us at our Discord channel.

For bugs 🐛, please open a ticket in Github."
`;
